crates/turborepo-lib/src/commands/mod.rs

```diff
@@ -55,18 +54,26 @@ impl CommandBase {
         })
     }
 
-    fn create_repo_config(&self) -> Result<()> {
+    fn repo_config_init(&self) -> Result<RepoConfig, ConfigError> {
         let repo_config_path = get_repo_config_path(self.repo_root.borrow());
 
-        let repo_config = RepoConfigLoader::new(repo_config_path)
+        RepoConfigLoader::new(repo_config_path)
             .with_api(self.args.api.clone())
             .with_login(self.args.login.clone())
             .with_team_slug(self.args.team.clone())
-            .load()?;
+            .load()
+    }
 
-        self.repo_config.set(repo_config)?;
+    pub fn repo_config(&self) -> Result<&RepoConfig, ConfigError> {
+        self.repo_config.get_or_try_init(|| self.repo_config_init())
+    }
```

Comment:
This is a rigid new pattern here across all of the config loading sources. It makes it super-clear what's happening each time.